addons/de/xrechnung/instructions.go

```diff
@@ -0,0 +1,107 @@
+package xrechnung
+
+import (
+	"github.com/invopop/gobl/bill"
+	"github.com/invopop/gobl/cbc"
+	"github.com/invopop/gobl/pay"
+	"github.com/invopop/validation"
+)
+
+// Payment keys for XRechnung SEPA direct debit and credit transfer
+const (
+	KeyPaymentMeansSEPACreditTransfer cbc.Key = "sepa-credit-transfer"
+	KeyPaymentMeansSEPADirectDebit    cbc.Key = "sepa-direct-debit"
+)
+
+var validPaymentKeys = []cbc.Key{
+	pay.MeansKeyCash,
+	pay.MeansKeyCheque,
+	pay.MeansKeyCreditTransfer,
+	pay.MeansKeyCard,
+	pay.MeansKeyDirectDebit,
+	pay.MeansKeyOther,
+	KeyPaymentMeansSEPACreditTransfer,
+	KeyPaymentMeansSEPADirectDebit,
+}
+
+// ValidatePaymentInstructions validates the payment instructions according to the XRechnung standard
+func ValidatePaymentInstructions(value interface{}) error {
+	inv, ok := value.(*bill.Invoice)
+	if !ok || inv == nil || inv.Payment == nil || inv.Payment.Instructions == nil {
```

Comment:
This doesn't look right. I'd expect the `ValidatePaymentInstructions` method to only validate a `pay.Instructions` instance.